## Whos the Greatest Basketball Player of All time? ##

In our household, we continue to debate whos the greatest, MJ, Lebron, Kobe, Bill Russell, etc
I decided to use data to informe the decision. 

Primary data source - https://www.nba.com/stats/alltime-leaders?StatCategory=PTS&SeasonType=Regular+Season&PerMode=PerGame

### Create a CSV file with key metrics for each player ###
I pulled all time, and playoff metrics, and saved into a file goat.csv

In [54]:
# Import csv file intoa pandas dataframe to conduct further analysis
import pandas as pd

data = pd.read_csv('goat.csv')
df = pd.DataFrame(data)
df


Player  Games  Seasons  PTS-all-time  Career-Pt-Avg  Career-AG  \
0            MJ   1072       15       32292.0           30.1        5.3   
1        Lebron   1368       20       37113.0           27.1        7.4   
2          Kobe   1346       20       33643.0           25.0        4.7   
3        Kareem   1560       20       38387.0           24.6        NaN   
4  Bill Russell    963       13           NaN            NaN        NaN   

   Career-RB  Career-FT-avg  MVP   FG%  ...  PlayoffPts-average  \
0        6.2           83.5  5.0  49.7  ...                33.4   
1        7.5           73.4  4.0  50.5  ...                28.7   
2        5.2           83.7  1.0  44.7  ...                25.6   
3        NaN           72.1  NaN  55.9  ...                 NaN   
4        NaN            NaN  NaN   NaN  ...                 NaN   

   Playoff-assist-average  Playoff-rebound-average  playoff-Ft-average  \
0                     5.7                      6.4                82.8   
1                     7.2                      9.0                74.0   
2                     4.7                      5.1                81.6   
3                     NaN                      NaN                 NaN   
4                     NaN                      NaN                 NaN   

   Unnamed: 16  Allstar-Games  Allstar-mvp  1st-team-all-nba  DPOY  ADT  
0          NaN           13.0          3.0              11.0   1.0  9.0  
1          NaN           18.0          3.0              18.0   0.0  5.0  
2          NaN           15.0          4.0               NaN   NaN  NaN  
3          NaN            NaN          NaN               NaN   NaN  NaN  
4          NaN            NaN          NaN               NaN   NaN  NaN  

[5 rows x 22 columns]

### Create difference columns, comparing players in each Row. ### 
MJ is top of list, and Lebron 2, so Lebrons data will be compared to MJ, Then Kobe to Lebron, etc.

In [56]:
#diff=df['Games'].diff()
df['game-diff']=df['Games'].diff()
df['season-diff']=df['Seasons'].diff()
df['pts-diff']=df['PTS-all-time'].diff()
df['ppg-diff']=df['Career-Pt-Avg'].diff()
df['mvp-diff']=df['MVP'].diff()
df['playoff-ppg-diff']=df['PlayoffPts-average'].diff()
df['allstar-mvp-diff']=df['Allstar-mvp'].diff()
df['DPOY-diff']=df['DPOY'].diff()
df['adt-diff']=df['ADT'].diff()

df

Player  Games  Seasons  PTS-all-time  Career-Pt-Avg  Career-AG  \
0            MJ   1072       15       32292.0           30.1        5.3   
1        Lebron   1368       20       37113.0           27.1        7.4   
2          Kobe   1346       20       33643.0           25.0        4.7   
3        Kareem   1560       20       38387.0           24.6        NaN   
4  Bill Russell    963       13           NaN            NaN        NaN   

   Career-RB  Career-FT-avg  MVP   FG%  ...  ADT  game-diff  season-diff  \
0        6.2           83.5  5.0  49.7  ...  9.0        NaN          NaN   
1        7.5           73.4  4.0  50.5  ...  5.0      296.0          5.0   
2        5.2           83.7  1.0  44.7  ...  NaN      -22.0          0.0   
3        NaN           72.1  NaN  55.9  ...  NaN      214.0          0.0   
4        NaN            NaN  NaN   NaN  ...  NaN     -597.0         -7.0   

   pts-diff  ppg-diff  mvp-diff  playoff-ppg-diff  allstar-mvp-diff  \
0       NaN       NaN       NaN               NaN               NaN   
1    4821.0      -3.0      -1.0              -4.7               0.0   
2   -3470.0      -2.1      -3.0              -3.1               1.0   
3    4744.0      -0.4       NaN               NaN               NaN   
4       NaN       NaN       NaN               NaN               NaN   

   DPOY-diff  adt-diff  
0        NaN       NaN  
1       -1.0      -4.0  
2        NaN       NaN  
3        NaN       NaN  
4        NaN       NaN  

[5 rows x 31 columns]

In [ ]:
#Convert results to csv file, and save to local drive
df.to_csv('goat-data-cleaned.csv')

* Lebron played 296 more games than Jordan
* Lebron played 5 more seasons than jordan
* Lebron has 4821 more total career points than Jordan
* Lebron averages (3.1 PPG) **less** PPG than Jordan
* Lebron has 1 **less** MVP than Jordan
* Lebron averages (4.7 PPG) **less** than Jordan in playoffs
* Lebron and Jordan are tied with 3 all star MVPs
* Lebron has **never** won Defensive Player of the Year, Jordan has won it once
* Lebron has 2 **fewer** Rings than Jordan
* Lebron has 4 **less** all NBA Defensive team appearances than Jordan
* Lebron has played in 5 more All star games than Jordan
* Lebron has made 7 more 1st team all NBA teams than Jordan
* Lebron has **not** won a college basketball championship. Jordan has won one